source/code/providers/Container_ContainerLog_Class_Provider.cpp

```diff
@@ -0,0 +1,288 @@
+/* @migen@ */
+#include <MI.h>
+#include "Container_ContainerLog_Class_Provider.h"
+
+#include <map>
+#include <stdio.h>
+#include <string>
+#include <string.h>
+#include <syslog.h>
+#include <time.h>
+#include <unistd.h>
+#include <uuid/uuid.h>
+#include <vector>
+
+#include "../cjson/cJSON.h"
+#include "../dockerapi/DockerRemoteApi.h"
+#include "../dockerapi/DockerRestHelper.h"
+
+#define LASTLOGQUERYTIMEFILE "/var/opt/microsoft/docker-cimprov/state/LastLogQueryTime.txt"
```

Comment:
Note that there is a class in the PAL that may be of assistance for that, see either the older, more complicated (but perhaps more robust) [persist](https://github.com/Microsoft/pal/tree/master/source/code/scxcorelib/util/persist) stuff, or for something much smaller and simpler, look at the [config](https://github.com/Microsoft/pal/blob/master/source/code/scxcorelib/pal/scxconfigfile.cpp) stuff, which would work for this as well.